pkg/bloomgateway/sharding.go

```diff
@@ -0,0 +1,162 @@
+package bloomgateway
+
+import (
+	"context"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/dskit/ring"
+)
+
+// TODO(chaudum): Replace this placeholder with actual BlockRef struct.
+type BlockRef struct {
+	FromFp, ThroughFp uint64
+	FromTs, ThroughTs int64
+}
+
+var (
+	// BlocksOwnerSync is the operation used to check the authoritative owners of a block
+	// (replicas included).
+	BlocksOwnerSync = ring.NewOp([]ring.InstanceState{ring.JOINING, ring.ACTIVE, ring.LEAVING}, nil)
+
+	// BlocksOwnerRead is the operation used to check the authoritative owners of a block
+	// (replicas included) that are available for queries (a bloom gateway is available for
+	// queries only when ACTIVE).
+	BlocksOwnerRead = ring.NewOp([]ring.InstanceState{ring.ACTIVE}, nil)
+
+	// BlocksRead is the operation run by the querier to query blocks via the bloom gateway.
+	BlocksRead = ring.NewOp([]ring.InstanceState{ring.ACTIVE}, func(s ring.InstanceState) bool {
+		// Blocks can only be queried from ACTIVE instances. However, if the block belongs to
+		// a non-active instance, then we should extend the replication set and try to query it
+		// from the next ACTIVE instance in the ring (which is expected to have it because a
+		// bloom gateway keeps their previously owned blocks until new owners are ACTIVE).
+		return s != ring.ACTIVE
+	})
+)
+
+type Limits interface {
+	BloomGatewayShardSize(tenantID string) int
+}
+
+type ShardingStrategy interface {
```

Comment:
nit: this also looks copy+pasted from the index-gw, can't we reuse?

Reply:
Yes, will consolidate both the ring manager scaffolding and the sharding strategy in a separate PR after merging this one.